docusaurus/docs/ui-extensions/create-an-extension-point.md

```diff
@@ -15,40 +15,271 @@ keywords:
+
+```tsx
+import { usePluginLinkExtensions } from '@grafana/runtime';
+
+// We are defining the `context` outside of the React component for performance reasons.
+// (In case we would declare it inside the component below, then it would result in a new object on every render,
+// which would unnecessarily trigger the `usePluginLinkExtensions()` hook.)
+const context = {
+  referenceId: '12345',
+  timeZone: 'UTC',
+};
+
+function AppMenuExtensionPoint() {
+  // This only returns type="link" extensions
+  const { extensions } = usePluginLinkExtensions({
+    extensionPointId: 'plugins/another-app-plugin/menu',
+    context,
+  });
+
+  if (extensions.length === 0) {
+    return null;
+  }
+
+  return (
+    <div>
+      {extensions.map((extension) => {
+        return (
+          <a href={extension.path} onClick={extension.onClick} title={extension.description} key={extension.key}>
+            {extension.title}
+          </a>
+        );
+      })}
+    </div>
+  );
+}
+```
+
+#### Example - rendering link extensions (dynamic context)
+
+In this example we are creating the context object dynamically. Although this is a common practice, we have to be aware, that the `usePluginLinkExtensions()` hook will rerender in the following scenarios:
+
+- IF the `context` object changes _(so the extensions can react to the context changes)_
+- IF the extension-registry changes
+
+Because of these we have to make sure, that we only change the `context` object if its content changes, otherwise we will cause unnecessary rerenders. The following example shows how to approach these scenarios in a safe way:
+
+```tsx
+import { useMemo } from 'react';
+import { usePluginLinkExtensions } from '@grafana/runtime';
+
+function AppMenuExtensionPoint({ referenceId }) {
+  // Instead of just defining the object here (which would result in a new object on every render),
+  // we use `useMemo()` to only update the context object when its "dynamic" dependencies change.
+  const context = useMemo(
+    () => ({
+      referenceId,
+      timeZone: 'UTC',
+    }),
+    [referenceId]
+  );
+  const { extensions } = usePluginLinkExtensions({
+    extensionPointId: 'plugins/another-app-plugin/menu',
+    context,
+  });
+
+  if (extensions.length === 0) {
+    return null;
+  }
+
+  return (
+    <div>
+      {extensions.map((extension) => {
+        return (
+          <a href={extension.path} onClick={extension.onClick} title={extension.description} key={extension.key}>
+            {extension.title}
+          </a>
+        );
+      })}
+    </div>
+  );
+}
+```
+
+#### Example - rendering component extensions
+
+:::note
+
+**Available in Grafana >=10.1.0** <br /> (_Component type extensions are only available in Grafana 10.1.0 and above._)
+
+:::
+
+Component type extensions are simple React components, which gives much more freedom on what they are able to do. You can pass contextual information to the extension components using props.
```

Comment:
```suggestion
Component type extensions are simple React components, which gives you much more freedom in  what you can make them do. For example, you can pass contextual information to the extension components using props.
```

Reply:
Thanks! I have fixed it in https://github.com/grafana/plugin-tools/pull/881/commits/a5821d379642c396fdd2163dcecb71afb89a7e9c, however I left out the `For example` part (I think this is more like an additional information than an example), please let me know if you disagree.